llama_index/chat_engine/types.py

```diff
@@ -24,13 +25,27 @@ def chat(
         """Main chat interface."""
         pass
 
+    @abstractmethod
+    def stream_chat(
+        self, message: str, chat_history: Optional[List[ChatMessage]] = None
+    ) -> Any:
```

Comment:
I see why the typing here is hard. 
Is it basically a Union[StreamingChatResponse, RESPONSE_TYPE]?

Reply:
We can also define a `Protocol` to describe the type here, which is basically something that has a `response_gen` and a `__str__` method?

Reply:
Hmm that might be a good idea. I can look into streamlining it this way

Reply:
Didn't use a protocol, but ended up fixing the typing another way :)